frontend/src/models/devices.ts

```diff
@@ -44,16 +44,13 @@ export default createModel({
     async shouldSearchDevices() {
       // First see if they have already decided on their preference
       const pref = window.localStorage.getItem(SEARCH_ONLY_SETTING_KEY)
-      if (typeof pref === 'string') {
-        dispatch.devices.setSearchOnly(pref === 'true')
-        dispatch.devices.fetch()
-        return
-      }
+      let searchOnly = pref !== 'false'
 
       // If they have too many services, show search only.
-      return r3.devices
-        .count()
-        .then(count => dispatch.devices.setSearchOnly(count.services > SEARCH_ONLY_SERVICE_LIMIT))
+      await r3.devices.count().then(count => (searchOnly = count.services > SEARCH_ONLY_SERVICE_LIMIT))
```

Comment:
Why not use await, assign count to var and then do your variable setting? Seems more idiomatic async/await

Reply:
Fixed!